servicetalk-http-api/src/main/java/io/servicetalk/http/api/MultiAddressHttpClientBuilder.java

```diff
@@ -336,4 +131,38 @@
      * @return {@code this}.
      */
     public abstract MultiAddressHttpClientBuilder<U, R> maxRedirects(int maxRedirects);
+
+    /**
+     * Builds a new {@link StreamingHttpClient}, using a default {@link ExecutionContext}.
+     *
+     * @return A new {@link StreamingHttpClient}
+     */
+    public abstract StreamingHttpClient buildStreaming();
```

Comment:
I considered adding a new interface just for these actual build methods. For now I went with copying, but if it makes sense, those methods could reside as default implementations in a new interface that is implemented by all other client builders. I'd appreciate opinions whether that makes sense.

Reply:
Would the interface be just for the default implementations or would the interface be used in API, field or variable types?

Reply:
HttpClientBuildFinalizer.java:
```
interface HttpClientBuildFinalizer {
    StreamingHttpClient buildStreaming();
    
    default HttpClient build() {
        return buildStreaming().asClient();
    }
    
    default BlockingStreamingHttpClient buildBlockingStreaming() {
        return buildStreaming().asBlockingStreamingClient();
    }
    
    default BlockingHttpClient buildBlocking() {
        return buildStreaming().asBlockingClient();
    }
}
```

used in MultiAddressHttpClientBuilder.java:

```
public abstract class MultiAddressHttpClientBuilder<U, R> implements HttpClientBuildFinalizer {
    // ...
}
```

implemented method in DefaultMultiAddressUrlHttpClientBuilder.java:
```
@Override
    public StreamingHttpClient buildStreaming() {
        // ...
    }
```

For other builders I believe it would be the same. So no, no exposure apart from the methods that are already there with those same signatures. Only binary compatibility breakage from this change's perspective.